cmd/yurt-tunnel-server/app/options/options.go

```diff
@@ -152,6 +157,55 @@ func (o *ServerOptions) Config() (*config.Config, error) {
 	}
 	cfg.SharedInformerFactory = informers.NewSharedInformerFactory(cfg.Client, 10*time.Second)
 
+	if cfg.ServerCount <= 0 {
+		cnt, err := guessServerCount(cfg.Client)
+		if err != nil {
+			klog.Warningf("failed to guess server count, set server count as 1, %v", err)
+			cfg.ServerCount = 1
+		} else {
+			klog.Infof("guess and set server count as %v", cnt)
+			cfg.ServerCount = cnt
+		}
+	}
+
 	klog.Infof("yurttunnel server config: %#+v", cfg)
 	return cfg, nil
 }
+
+// guessServerCount returns the replicas of the tunnel server pod
+// owner ReplicaSet if that is available
+func guessServerCount(client kubernetes.Interface) (int, error) {
```

Comment:
guessServerCount is not a good name.  How about tryObtainServerCountFromReplicaSet?

Reply:
Updated!